shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Examples/Combinators.hs

```diff
@@ -0,0 +1,217 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- |
+-- Module      : Test.Shelley.Spec.Ledger.Examples.Federation
+-- Description : Chain State Combinators
+--
+-- A collection of combinators for manipulating Chain State.
+-- The idea is to provide a clear way of describing the
+-- changes to the chain state when a block is processed.
+module Test.Shelley.Spec.Ledger.Examples.Combinators
+  ( -- | = Evolve Nonces - Frozen
+    --
+    -- Evolve the appropriate nonces under the assumption
+    -- that the candidate nonce is now frozen.
```

Comment:
I think this haddock should be on the function definitions rather than their exports - I'm not sure whether documenting here is valid haddock?

Reply:
it's valid haddock, I was building them and looking at them. I'll see what it looks like if I move them to the function definitions.

Reply:
I can't get the headers and sections to look like I want when I move the comments to the function definitions. if you don't mind, I'll merge now to get the out of the way in master, and then try to find something that looks good.